Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,100 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+
         // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        public IEnumerable<string> CsvRows { get; }
 
         // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            List<string> states = CsvRows.
+                Select(person => person.Split(',')[6]).
+                Distinct().
+                OrderBy(state => state).
+                ToList();

```

Comment:
```suggestion
            List<string> states = CsvRows
                .Select(person => person.Split(',')[6])
                .Distinct()
                .OrderBy(state => state)
                .ToList();
```
Good question! The `.` typically goes at the beginning of a new LINQ statement. This is because it makes it easy remove/rearrange/add statements without touching the lines above/below too much. Also a bit easier to see the `.` when it goes at the start of the statement.

Reply:
Additionally, you don't need to convert this to a list here. In some cases, it's actually better to not do so - https://www.tutorialsteacher.com/linq/linq-deferred-execution